tests/drivers/gpio/gpio_basic_api/src/main.c

```diff
@@ -115,6 +177,12 @@ static void board_setup(void)
 
 	pinmux_pin_set(pmx, PIN_OUT, PORT_PCR_MUX(kPORT_MuxAsGpio));
 	pinmux_pin_set(pmx, PIN_IN, PORT_PCR_MUX(kPORT_MuxAsGpio));
+#elif defined(CONFIG_GPIO_SIM)
```

Comment:
Use a local `extern struct gpio_callback *gpio_sim_callback;`  declaration to expose the content from the separate source file.